src/Chapter3.hs

```diff
@@ -910,8 +986,21 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
 
-{-
+instance Append Gold where
+  append (Gold g) (Gold h) = Gold $ g+h
+
+instance Append [a] where
+  append l1 l2 = l1 ++ l2
+
+instance (Append a) => Append (Maybe a) where
+  append (Just m1) (Just m2) = Just $ append m1 m2
+  append (Just m1) _ = Just m1
+  append _ (Just m2) = Just m2
+  append _ _ = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂